src/app/cluster/cluster-details/constraints/violation-details/component.ts

```diff
@@ -10,25 +10,40 @@
 // limitations under the License.
 
 import {Component, Input, OnInit, ViewChild} from '@angular/core';
+import {MatPaginator} from '@angular/material/paginator';
 import {MatSort} from '@angular/material/sort';
 import {MatTableDataSource} from '@angular/material/table';
 import {Violation} from '@shared/entity/opa';
+import {UserSettings} from '@shared/entity/settings';
 
 @Component({
   selector: 'km-violation-details-list',
   templateUrl: './template.html',
   styleUrls: ['./style.scss'],
 })
 export class ViolationDetailsComponent implements OnInit {
-  @Input() violations: Violation[];
+  @Input() violations: Violation[] = [];
+  @Input() settings: UserSettings;
   displayedColumns: string[] = ['name', 'enforcement', 'kind', 'message'];
   dataSource = new MatTableDataSource<Violation>();
   @ViewChild(MatSort, {static: true}) sort: MatSort;
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
 
   ngOnInit(): void {
     this.dataSource.data = this.violations || [];
     this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
```

Comment:
Paginator for Violations is a bit harder. It is there but the page will be resetted every few minutes due to "refresh" (e.g. you are on page 2, but it refreshes to 1 fairly quick). I've tried some other implementations, but nothing seems to work in the way it should be, so I just committed "simple" way for now. 
Anyhow, do you have an idea how to solve this page/refresh issue?

Reply:
I'd say that if the number of pages increases then there is no problem and we stay on the same page. If the number is decreased and the page that we were on will no longer exist then we should go to the last possible page. Does it make sense? Can it be implemented in that way?

Reply:
Sorry I somehow missed your comment. 
I'm not entirely sure if we're thinking the same 🤔 

Screenshot is a bit small but it should be enough to see the problem. The toggled area is refreshed every few seconds and therefor the paginator is set to 0 again and again. 
![pag](https://user-images.githubusercontent.com/19547196/112620969-79036100-8e29-11eb-961e-bce1264f6a02.gif)

I spend some additional time on it and tried a few different implementations and maybe I have done something wrong, but I didn't manage to get everything up and running. :/

Reply:
@maciaszczykm @floreks

Reply:
The current issue is that the `ViolationDetailsComponent` gets initialized each 5-10 seconds (add some log in the `ngOnInit` to see this). I'll check if I have any idea how to fix that.

Reply:
Sorry, I didn't have enough time yesterday. The workaround might be moving the pagination init somewhere else however I am not sure if it will work. Could you try that?

Reply:
You could also keep somewhere page index and set it during validation.

Reply:
Sorry, I didn't have enough time yesterday. The workaround might be moving the pagination init somewhere else or keeping the page index somewhere else and setting it during the init. I'm not sure if any of these will work though, could you try that?

Reply:
saving the index seems to work. PTAL @maciaszczykm

Reply:
Great, I'll review today :)